### PR TITLE
Big factors plus irm_test plus integration test of transition_hyperparameters

### PR DESCRIPTION
**Pull request by ThomasColthurst**
1) :hirm is broken up into :irm, :hirm_lib and :hirm.
2) :headers is removed
3) :relation_variant is split off from :util_distribution_variant in order to avoid cyclic dependencies.
4) A pretty small irm_test is added.
5) test_irm_two_relations.cc is upgraded to use single_step_irm_inference, and because of that must set the distribution hyperparameters from the IRM that is loaded from disk.
6) Fixed an unused variable warning in relation_test.cc

### CODE REVIEW

File: cxx/util_distribution_variant.hh [hunk 1]
```diff
@@ -1,8 +1,9 @@
 // Copyright 2024
 // See LICENSE.txt
 
-// This file collects classes/functions that depend on the set of distribution
-// subclasses and should be updated when a new subclass is added.
+// Classes and functions for dealing with Distributions and their values in a
```
Comment (emilyfertig) [inline cxx/util_distribution_variant.hh:4]: Could you update the comment on lines 7-8 of the Distribution base class too?

File: cxx/irm_test.cc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+#define BOOST_TEST_MODULE test IRM
+
+#include "irm.hh"
+#include "util_distribution_variant.hh"
+
+#include <boost/test/included/unit_test.hpp>
+namespace tt = boost::test_tools;
+
+BOOST_AUTO_TEST_CASE(test_irm) {
```
Comment (emilyfertig) [inline cxx/irm_test.cc:11]: Could you add a test for logp? Maybe something like this (assuming I understand correctly):

```
double logp_score0 = irm.logp_score();
double logp1 = irm.logp(x);
irm.incorporate(x);
double logp_score1 = irm.logp_score();
BOOST_TEST(logp_score1 == logp_score0 + logp1);
```
(Or fine to do in a follow-up, if you or Srinivas are planning on expanding test coverage/adding HIRM tests later.)

**Review by emilyfertig — APPROVED** [review]
Looks great, thank you!!

File: cxx/irm_test.cc (hunk 2 above)
Comment (ThomasColthurst) [reply cxx/irm_test.cc:11]: Well, I added that test, which I also expected to pass, but it didn't.  Added a TODO for now.

**Pull request merged**